sample-notificator-domain/src/main/groovy/life/qbic/business/notification/create/CreateNotification.groovy

```diff
@@ -1,30 +1,167 @@
     }
 
     @Override
     void createNotifications(String date) {
-        LocalDate localDate = LocalDate.parse(date)
-        List<Subscriber> subscribers = ds.getSubscribersForNotificationsAt(localDate)
-        println subscribers
-        //todo send notifications and fill template
+        fetchSubscriberInput.fetchSubscriber(date)
+        output.createdNotifications(notificationPerSubscriber)
+    }
+
+    /**
+     * Method which returns a map containing the SAMPLE_QC_FAIL notification for each subscriber
+     * @param subscribers list of subscribers, which are to be informed about a change in SampleStatus
+     * @return map associating the generated notification with the to be informed subscriber
+     */
+    private static Map<Subscriber, String> createNotificationPerSubscriber(List<Subscriber> subscribers) {
+
+        Map<Subscriber, String> notificationPerSubscriber = [:]
+        subscribers.each {
+            List<String> sampleCodes = getSampleCodesFromSubscriber(it)
+            String failedNotification = failedQCMailMessage(it, sampleCodes)
+            notificationPerSubscriber[it] = failedNotification
+        }
+
+        return notificationPerSubscriber
+    }
+
+    /**
+     * Method which extracts the SampleCodes set to SAMPLE_QC_FAIL for a given subscriber
+     * @param subscriber Subscriber which is to be notified about SAMPLE_QC_FAIL Sample Codes
+     * @return List of SampleCodes whose status is set to SAMPLE_QC_FAIL
+     */
+    private static List<String> getSampleCodesFromSubscriber(Subscriber subscriber) {
+        List<String> failedQCCodes = []
+        subscriber.subscriptions.each {
+            if(it.value.toString() == "SAMPLE_QC_FAIL") {
+                failedQCCodes.add(it.key)
+            }}
+        return failedQCCodes
+    }
+
+    /**
+     * Method which assembles the email Template informing a subscriber about the projects containing SAMPLE_QC_FAIL samples
+     * @param subscriber Subscriber for whom the template email is generated
+     * @param failedQCSampleCodes List of SampleCodes whose status were set to SAMPLE_QC_FAIL.
+     * @return String containing the assembled SAMPLE_QC_FAIL template email message.
+     */
+    private static String failedQCMailMessage(Subscriber subscriber, List<String> failedQCSampleCodes) {
+
+        String notificationIntro =
+                """
+                Dear ${subscriber.firstName} ${subscriber.lastName},
+                this is an automated email to inform you that samples in the following projects have failed the quality control step:
+                <a href="Link to some explanation">Click here to learn more.</a>
+                """.stripIndent()
+
+        String projectList = listProjects(failedQCSampleCodes)
+
+        String notificationOutro =
+                """
+                If you would like to unsubscribe from this or other project updates, you can do so by clicking <a href="Link to subscription thingy">here</a>.
```

Comment:
For now this was not Part of the card, but it would make sense to have an unsubscribe mechanism somewhere. I think this can be delayed until we get feedback from the Stakeholders concerning the template message itself

Reply:
makes sense